### PR TITLE
Add Core Workflow documentation

### PR DESCRIPTION
**Pull request by MrGeneration**
This PR provides a new section for core workflow. 
It also rather drastically reworks the object section which now has a better structure and greater details.

### CODE REVIEW

File: system/core-workflow/how-do-they-work.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+     attributes. We're displaying the following icons for your overview
+     to understand these limits better. 👀
+   | |api| This icon indicates the action affects the API.
+   | |ui| This Icon indicates the action only affects the web interface.
+
+show |ui|
+   Display the field in question. Allows setting of values.
+
+hide |ui|
+   Hide the field in question, however,
```
Comment (techerella) [inline system/core-workflow/how-do-they-work.rst:114]: however, this technically...

File: system/core-workflow/how-do-they-work.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+set optional |ui| |api|
+   Sets the field to optional.
+
+add option |ui| |api|
+   Allows adding options to tree selects or selects.
+
+      .. note::
+
+         This requires options to be hidden before hand (remove option).
```
Comment (techerella) [inline system/core-workflow/how-do-they-work.rst:141]: beforehand

File: system/core-workflow/how-do-they-work.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+   Select a specific value within a select, tree select or boolean fields.
+
+auto select |ui|
+   | Helps the user on tree selects and select fields:
+   | If the field has one option to select only and has no value yet, the
+     value is automatically set.
+
+   .. warning::
+
+      This option only works if you have one value and acts passive with more
```
Comment (techerella) [inline system/core-workflow/how-do-they-work.rst:176]: is passive / acts passively

File: system/core-workflow/how-do-they-work.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+   :width: 42px
+
+.. |ui| image:: /images/icons/ui-symbol.png
+   :height: 42px
+   :width: 42px
+
+Stop after match
+----------------
+
+Stop evaluation of other, following workflows that would match other wise.
```
Comment (techerella) [inline system/core-workflow/how-do-they-work.rst:197]: otherwise

File: system/core-workflow/how-do-they-work.rst [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+Default: ``no``
+
+Priority
+--------
+
+By adjusting the priority you decide when a workflow is checked on.
```
Comment (techerella) [inline system/core-workflow/how-do-they-work.rst:204]: maybe: "when a workflow is activated"?

File: system/core-workflow/learn-by-example.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
+Learn by example
+================
+
+This page provides some of the ideas we had for Core Workflow.
```
Comment (techerella) [inline system/core-workflow/learn-by-example.rst:4]: WorkflowS

File: system/core-workflow/learn-by-example.rst [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+Of course you can built much more complex workflows.
```
Comment (techerella) [inline system/core-workflow/learn-by-example.rst:5]: build

File: system/core-workflow/learn-by-example.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+               This reduces the category options to ``Sales/*``, 
+               ``Internal/*`` and ``Others``.
+
+               .. figure:: /images/system/core-workflow/examples/1_group-specific-fields-and-values_sales.png
+                  :alt: Sample workflow that shows specific values and fields for sales
+                  :width: 90%
+
+         The Result
+            This is what the agent would experience with above
```
Comment (techerella) [inline system/core-workflow/learn-by-example.rst:62]: with the above workflows

File: system/core-workflow/learn-by-example.rst [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+            workflows in place.
+            .. figure:: /images/system/core-workflow/examples/1_group-specific-fields-and-values_result.gif
+            :alt: Workflow shows objects and limits options based on selections on the group
+            :width: 90%
+
+2. Approval process
+      In this case ``aproved`` is visible to agents by default.
```
Comment (techerella) [inline system/core-workflow/learn-by-example.rst:69]: approved

File: system/core-workflow/learn-by-example.rst [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+      For this workflow a additional role ``Approval person`` is required
```
Comment (techerella) [inline system/core-workflow/learn-by-example.rst:70]: ...workflow, an additional role....

File: system/core-workflow/learn-by-example.rst [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+      (no further permissions).
+
+      .. figure:: /images/system/core-workflow/examples/2_role-specific-approval-settingsl.png
+         :alt: Sample workflow that restricts an approval attribute to specific roles
+         :width: 90%
+
+      .. tip::
+
+         This workflow may work best in combination with a 
+         :doc:`trigger </manage/trigger>` but technically is not required.
```
Comment (techerella) [inline system/core-workflow/learn-by-example.rst:80]: technically, this is...

File: system/core-workflow/limitations.rst [hunk 12]
```diff
@@ -0,0 +1,36 @@
+Limitations
+===========
+
+Core Workflow does not replace Trigger
```
Comment (techerella) [inline system/core-workflow/limitations.rst:4]: WorkflowS

File: system/core-workflow/limitations.rst [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+  Workflows manipulate behavior of fields, however, does not set values
```
Comment (techerella) [inline system/core-workflow/limitations.rst:5]: manipulate the behavior of fields, however, they do not set values...

File: system/core-workflow/limitations.rst [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+  This affects the following actions:
+
+    * select
+    * auto select
+    * show
+    * hide
+
+Some fields stay unavailable to customers
+  For technical and security reasons some default fields (the greyed out ones
```
Comment (techerella) [inline system/core-workflow/limitations.rst:19]: reasons,

Comment (techerella) [inline system/core-workflow/limitations.rst:19]: the pale ones

File: system/core-workflow/limitations.rst [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+  you can't edit) stay unavailable for display and usage on customer
+  permissions.
+
+    .. hint::
+
+      If you require your customers to change e.g. priorities, please
+      consider using workarounds via :doc:`/system/objects` and
+      :doc:`/manage/trigger`.
+
+What is out of scope of Core Workflow...?
```
Comment (techerella) [inline system/core-workflow/limitations.rst:29]: WorkflowS

File: system/objects.rst [hunk 16]
```diff
@@ -1,119 +1,75 @@
...
-This can be useful if you need to add further information to a ticket that you don't want to have in a note (you'll find it easier).
+In Zammad you can add your own fields to tickets, users, organizations and even
+groups. This can be useful if you need to add further information to a ticket
+that you don't want to have in a note (you'll find it easier).
 
-.. note:: Try to avoid deleting objects (rather disable them) as Zammad might run into unexpected conditions if they where referenced somewhere.
+.. note:: 
 
-.. image:: /images/system/objects/1.jpg
+   Try to avoid deleting objects (rather disable them) as Zammad might run into
```
Comment (techerella) [inline system/objects.rst:10]: (and disable them instead)

File: system/objects.rst [hunk 17]
```diff
@@ -1,119 +1,75 @@
...
+   unexpected conditions if they where referenced somewhere.
```
Comment (techerella) [inline system/objects.rst:11]: conditions if they are referenced somewhere

File: system/objects.rst [hunk 18]
```diff
@@ -1,119 +1,75 @@
...
-
-``https://www.google.com/search?q=#{ticket.amazingobject}``
-
-The above will create a google search link which will contain the search phrase (which is the objects value).
-You can use any variable that's available within Zammad! This gives your great options within your structures!
+Here's an overview of the objects. On the upper right you can add new
+Attributes (no 1). By default, there will be no custom fields - standard objects
+(no 2) will be grayed out, you can't delete or change those.
+Custom objects (no 3) will be displayed in black font and have a trash bin on
+the right site to delete not needed objects.
```
Comment (techerella) [inline system/objects.rst:19]: unnecessary objects

File: system/objects.rst [hunk 19]
```diff
@@ -1,119 +1,75 @@
...
+By click on custom objects, you can edit them so they can suite your needs.
```
Comment (techerella) [inline system/objects.rst:20]: By clicking on "Custom Objects" you can edit them to suit your needs

File: system/objects.rst [hunk 20]
```diff
@@ -1,119 +1,75 @@
...
+to Zammad. If you made a mistake or just want to discard your changes, click
+"Discard changes".
 
-Object permissions
-------------------
+.. warning::
 
-When ever needed you can restrict access to objects for permission roles (agents and customer) and even set a object to be required.
+   After applying the object changes with "Update Database" a restart of Zammad
+   is **mandatory**. If you don't you may experience unexpected behavior or
```
Comment (techerella) [inline system/objects.rst:41]: If you don't perform it, you...

File: system/objects.rst [hunk 21]
```diff
@@ -1,119 +1,75 @@
...
+         Changes on objects require you to update the database
+         to apply these changes.
 
-.. figure:: /images/system/objects/2.jpg
-   :align: center
+   .. tip:: **🤓 Service restarts can be automated**
 
-   This is the object edit screen, it looks very similar to the creation screen and holds all object specific information.
+      | Hosted environments automatically restart for you.
+      | If you're using a self hosted installation you can use
```
Comment (techerella) [inline system/objects.rst:54]: self-hosted

File: system/objects.rst [hunk 22]
```diff
@@ -1,119 +1,75 @@
...
 
+.. _environment variables:
+   https://docs.zammad.org/en/latest/appendix/configure-env-vars.html
 
+System objects
+--------------
 
-Updating database after adding or editing objects
--------------------------------------------------
+Zammad comes with pre shipped objects. Some of these currently do not
```
Comment (techerella) [inline system/objects.rst:63]: pre-configured?

File: system/objects/permissions.rst [hunk 23]
```diff
@@ -0,0 +1,71 @@
+Object permissions
+******************
+
+.. figure:: /images/system/objects/permission-and-screen-overview.png
+   :align: center
+
+   Some of the possible permission and screen options for objects.
+
+When ever needed you can restrict access to objects based on the
```
Comment (techerella) [inline system/objects/permissions.rst:9]: Whenever needed,

File: system/objects/permissions.rst [hunk 24]
```diff
@@ -0,0 +1,71 @@
...
+.. figure:: /images/system/objects/permission-and-screen-overview.png
+   :align: center
+
+   Some of the possible permission and screen options for objects.
+
+When ever needed you can restrict access to objects based on the 
+:ref:`user permission <permission-guide>`
+(``admin``, ``ticket.agent`` & ``ticket.customer``).
+
+.. tip:: **🤓 Below is not written to stone 🤓**
```
Comment (techerella) [inline system/objects/permissions.rst:13]: set in stone

File: system/objects/permissions.rst [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+
+   You can always adjust below settings with :doc:`/system/core-workflow`.
+   This also allows role based restriction. 👀
+
+.. note::
+   
+   In some situation Zammad internally overrules below screen, requirement and
```
Comment (techerella) [inline system/objects/permissions.rst:20]: situations,

File: system/objects/permissions.rst [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+            * Organization
+            * Group
+
+invite_customer & invite_agent
+   Shown when using the invitation dialogue from "First Steps" in the dashboard.
+
+About screen options
+--------------------
+
+Now we know the different possible situations,
```
Comment (techerella) [inline system/objects/permissions.rst:63]: Now that we know

File: system/objects/types.rst [hunk 27]
```diff
@@ -0,0 +1,166 @@
+Object types
+************
+
+When adding a new object, you can choose between the following object types.
+
+Boolean
+   Provides a drop down field with display values for ``true`` and ``false``.
```
Comment (techerella) [inline system/objects/types.rst:7]: drop-down

File: system/objects/types.rst [hunk 28]
```diff
@@ -0,0 +1,166 @@
...
+         `bug 3763 <https://github.com/zammad/zammad/issues/3763>`_.
+
+   .. figure:: /images/system/objects/settings_date.png
+      :alt: Available settings for Date fields
+
+Datetime
+   Provides a date and time picker – does not allow default values
+
+   Allow future
+      | Allows to disallow dates and times in the future.
```
Comment (techerella) [inline system/objects/types.rst:32]: forbid dates and times

File: system/objects/types.rst [hunk 29]
```diff
@@ -0,0 +1,166 @@
...
+      | Default: ``yes``
+
+   Allow past
+      | Allows to disallow dates and times in the past.
```
Comment (techerella) [inline system/objects/types.rst:36]: forbid dates and times

File: system/objects/types.rst [hunk 30]
```diff
@@ -0,0 +1,166 @@
...
+      .. note::
+
+         Please note that this setting is currently affected by
+         `bug 3763 <https://github.com/zammad/zammad/issues/3763>`_.
+
+   .. figure:: /images/system/objects/settings_datetime.png
+      :alt: Available settings for Date time fields
+
+Integer
+   Provides a input field that allows integer usage only.
```
Comment (techerella) [inline system/objects/types.rst:52]: an input field

File: system/objects/types.rst [hunk 31]
```diff
@@ -0,0 +1,166 @@
...
+      :alt: Available settings for Integer fields
+
+Select
+   Provides a drop down field that allows selection of one value out of several.
+   This field does allow setting a default value.
+
+   .. tip::
+
+      Adding values can be tricky for first timers, don't forget to press
+      "➕ Add" after typing your values. Other wise you may loose a value.
```
Comment (techerella) [inline system/objects/types.rst:71]: Otherwise you may lose

File: system/objects/types.rst [hunk 32]
```diff
@@ -0,0 +1,166 @@
...
+   .. hint::
+
+      Select fields allow usage of
+      :ref:`URL fields (Link Templates) <link-templates>`.
+
+   .. figure:: /images/system/objects/settings_text.png
+      :alt: Available settings for Text fields
+
+Tree Select
+   Provides a select like field with up to 6 layers of options.
```
Comment (techerella) [inline system/objects/types.rst:108]: select-like

File: system/objects/types.rst [hunk 33]
```diff
@@ -0,0 +1,166 @@
...
+URL fields (Link-Template)
+--------------------------
+
+.. note:: 
+
+   This function is restricted to Text and Select objects only.
+
+Link-Templates are an amazing way to dynamically generate URLs.
+They allow you to integrate other systems better without having to
+manually copying data from Zammad if possible.
```
Comment (techerella) [inline system/objects/types.rst:130]: manually copy

File: system/objects/types.rst [hunk 34]
```diff
@@ -0,0 +1,166 @@
...
+
+   .. note::
+
+      Another great way of communicating with another system may be Zammads
```
Comment (techerella) [inline system/objects/types.rst:134]: Zammad's

File: system/objects/types.rst [hunk 35]
```diff
@@ -0,0 +1,166 @@
...
+      :doc:`Webhooks </manage/webhook>`.
+
+After filling a link-template enabled fields a URL icon will appear on its
```
Comment (techerella) [inline system/objects/types.rst:137]: field, a URL

File: system/objects/types.rst [hunk 36]
```diff
@@ -0,0 +1,166 @@
...
+right. Clicking on the icon opens a new tab.
+
+   .. hint:: 
+
+      Even through Zammad displays the link template within object edit and
```
Comment (techerella) [inline system/objects/types.rst:142]: Even though

File: system/objects/types.rst [hunk 37]
```diff
@@ -0,0 +1,166 @@
...
+      create screens the function is optional. It's only active if you populate
```
Comment (techerella) [inline system/objects/types.rst:143]: screens, the

File: system/objects/types.rst [hunk 38]
```diff
@@ -0,0 +1,166 @@
...
+      the field.
+
+How does this work...?!
+^^^^^^^^^^^^^^^^^^^^^^^
+
+As example, let's say you have an object called ``amazingobject`` - you want
```
Comment (techerella) [inline system/objects/types.rst:149]: An an example

**Review by techerella — APPROVED** [review]

File: system/core-workflow/how-do-they-work.rst (hunk 5 above)
Comment (MrGeneration) [reply system/core-workflow/how-do-they-work.rst:204]: Doesn't fit in my opinion.
This part is about the user being able to adjust the priority of the workflows as needed.

So the order in which they are evaluated in.
While that's only relevant when they're active, the moment you adjust this value doesn't matter.

This screenshot may help (lower end)
https://github.com/zammad/zammad-admin-documentation/blob/mh/add/core-workflow/images/system/core-workflow/add-new-workflow.png

Comment (techerella) [reply system/core-workflow/how-do-they-work.rst:204]: Ah okay! That wasn't quite clear to me through the above wording, but perhaps that's just me. Perhaps we could rephrase to "you decide at which point the workflow is checked on", but I'll leave it up to you

Comment (MrGeneration) [reply system/core-workflow/how-do-they-work.rst:204]: Okay, 
I switched this from `By adjusting the priority you decide when a workflow is checked on.`
to `You decide at which point your workflow is evaluated.` hoping it fits well.

**Comment by MrGeneration** [comment]
Hey @rolfschmidt - could you please also have a look to see if everything is there in your opinion?
Anything missing? Any miss leads?

**Review by rolfschmidt — APPROVED** [review]

**Pull request merged**